notebook/static/tree/js/treeactions.js

```diff
@@ -0,0 +1,322 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+define(function(require){
+    "use strict";
+
+    var ActionHandler = function (env) {
+        this.env = env || {};
+        Object.seal(this);
+    };
+
+    /**
+     *  A bunch of predefined `Simple Actions` used by Jupyter.
+     *  `Simple Actions` have the following keys:
+     *  help (optional): a short string the describe the action.
+     *      will be used in various context, like as menu name, tool tips on buttons,
+     *      and short description in help menu.
+     *  help_index (optional): a string used to sort action in help menu.
+     *  icon (optional): a short string that represent the icon that have to be used with this
+     *  action. this should mainly correspond to a Font_awesome class.
+     *  handler : a function which is called when the action is activated. It will receive at first parameter
+     *      a dictionary containing various handle to element of the notebook.
+     *
+     *  action need to be registered with a **name** that can be use to refer to this action.
+     *
+     *
+     *  if `help` is not provided it will be derived by replacing any dash by space
+     *  in the **name** of the action. It is advised to provide a prefix to action name to
+     *  avoid conflict the prefix should be all lowercase and end with a dot `.`
+     *  in the absence of a prefix the behavior of the action is undefined.
+     *
+     *  All action provided by Jupyter are prefixed with `ipython.`.
+     *
+     *  One can register extra actions or replace an existing action with another one is possible
+     *  but is considered undefined behavior.
+     *
+     **/
+    var _actions = {
+        'enter-row': {
+            help    : 'enters row',
+            help_index : 'aa',
+            handler : function (env) {
+                var len_sib = env.notebooklist.get_length_siblings('.focus');
+                if (len_sib > 0){
+                  env.notebooklist.open();
+                }
+            }
+        },
+        'select-previous-row':{
+            help    : 'selects previous row',
```

Comment:
On the notebook list, 'select' to me means checking the checkbox next to an item. I'd describe what this does in terms of a 'cursor' or 'focus', e.g. 'move cursor down'.